components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -446,6 +454,18 @@ impl FileLoader {
         })
     }
 
+    pub fn read<T: serde::de::DeserializeOwned>(&self, file: &FilePath) -> Result<T> {
+        let string = self
+            .read_to_string(file)
+            .map_err(|e| FMLError::InvalidPath(format!("{file}: {e}")))?;
+
+        Ok(match file.extension().map(str::to_lowercase).as_deref() {
```

Comment:
Why do you need the `.as_deref()` here?

Reply:
We want to map an `Option<String>` to an `Option<&String>`, and thence to a `Option<&str>`. `as_deref()` does this for us.

Reply:
Ahh that's cool 👍